File: examples/finatra-hello-world-heroku/src/main/scala/com/twitter/hello/HelloWorldServer.scala
package com.twitter.hello

import com.twitter.finagle.http.{Response, Request}
import com.twitter.finatra.http.HttpServer
import com.twitter.finatra.http.filters.CommonFilters
import com.twitter.finatra.http.routing.HttpRouter
import com.twitter.finatra.logging.filter.{TraceIdMDCFilter, LoggingMDCFilter}
import com.twitter.finatra.logging.modules.LogbackModule

object HelloWorldServerMain extends HelloWorldServer

class HelloWorldServer extends HttpServer {
  override def modules = Seq(LogbackModule)

  override def configureHttp(router: HttpRouter) {
    router
      .filter[LoggingMDCFilter[Request, Response]]
      .filter[TraceIdMDCFilter[Request, Response]]
      .filter[CommonFilters]
      .add[HelloWorldController]
  }
}

File: examples/java-http-server/src/main/java/com/twitter/hello/server/HelloWorldServer.java
package com.twitter.hello.server;

import java.util.Collection;
import java.util.Collections;

import scala.reflect.ManifestFactory;

import com.google.inject.Module;

import com.twitter.app.Flaggable;
import com.twitter.finatra.http.AbstractHttpServer;
import com.twitter.finatra.http.filters.CommonFilters;
import com.twitter.finatra.http.routing.HttpRouter;
import com.twitter.hello.server.controllers.HelloWorldController;
import com.twitter.hello.server.exceptions.HelloWorldExceptionMapper;
import com.twitter.hello.server.filters.AppendToHeaderFilter;
import com.twitter.hello.server.modules.MagicNumberModule;

public class HelloWorldServer extends AbstractHttpServer {

  public HelloWorldServer() {
    createFlag(
        /* name = */ "magic.number",
        /* default = */ 55,
        /* help = */ "This is a magic number.",
        /* flaggable = */ Flaggable.ofJavaInteger());
  }

  @Override
  public Collection<Module> javaModules() {
    return Collections.singletonList(new MagicNumberModule());
  }

  @Override
  public void configureHttp(HttpRouter router) {
    router
        .filter(CommonFilters.class)
        .filter(new AppendToHeaderFilter("foo", "1"))
        .add(HelloWorldController.class)
        .register(ManifestFactory.classType(CatMessageBodyReader.class))
        .register(ManifestFactory.classType(DogMessageBodyWriter.class))
        .exceptionMapper(HelloWorldExceptionMapper.class);
  }
}
